### PR TITLE
Add register users gateway

### PR DESCRIPTION
**Pull request by StevenLeighton21**

### CODE REVIEW

File: spec/lib/hackney/income/register_users_gateway_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  let(:response) do
+    {
+      id: 123,
+      name: 'Richard Foster',
+      email: 'rich@richy.rich',
+      first_name: 'Richard',
+      last_name: 'Foster',
+      provider_permissions: true
+    }
```
Comment (zuren) [inline spec/lib/hackney/income/register_users_gateway_spec.rb]: Could do with randomisation here too, to guarantee the response isn't slimed?

File: lib/hackney/income/register_users_gateway.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
+module Hackney
+  module Income
+    class RegisterUsersGateway
```
Comment (zuren) [inline lib/hackney/income/register_users_gateway.rb]: Feel like `IncomeApiUsersGateway` might be more fitting? Really easy to see what it touches that way, similar to `SqlUsersGateway`.

File: lib/hackney/income/register_users_gateway.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        params = {
+          provider_uid: provider_uid,
+          provider: provider,
+          name: name,
+          email: email,
+          first_name: first_name,
+          last_name: last_name,
+          provider_permissions: provider_permissions
+        }
+        uri.query = URI.encode_www_form(params)
```
Comment (zuren) [inline lib/hackney/income/register_users_gateway.rb]: Might be quicker to read without the `params` assignment?

Comment (StevenLeighton21) [reply lib/hackney/income/register_users_gateway.rb]: How do you mean? From a human/code maintenance point of view or optimisation?

Comment (zuren) [reply lib/hackney/income/register_users_gateway.rb]: From a human perspective. Up to you really, I just find the below easier to read, as there are fewer variables. 🤷‍♀️ 

```
uri.query = URI.encode_www_form(
  provider_uid: provider_uid,
  provider: provider,
  name: name,
  email: email,
  first_name: first_name,
  last_name: last_name,
  provider_permissions: provider_permissions
)
```

**Pull request merged**